server/src/cm_job_task.cpp

```diff
@@ -6971,32 +6999,226 @@ ts_get_tran_info (nvplist *req, nvplist *res, char *_dbmt_error)
 	}
       else
 	{
+	  int i;
+	  char *qbuf;
+
 	  nv_add_nvp (res, "SQL_ID", tok[8]);
 
-          sqltext = tok[8] + strlen (tok[8]) + 1;
-          while (sqltext && *sqltext)
-            {
-              if (*sqltext == ' ')
-                {
-                  sqltext++;
-                }
-              else
-                {
-                  break;
-                }
-            }
-          nv_add_nvp (res, "SQL_Text", sqltext);
+	  for (i = 0; i < num_queries; i++)
+	    {
+	      if (strncmp (sql_info[i].sql_id, tok[8], strlen (tok[8])) == 0)
+	        {
+	          qbuf = (char *) calloc (1, sql_info[i].len + 1);
+	          if (qbuf != NULL)
+	            {
+	              strncpy (qbuf, query_p + sql_info[i].offset, sql_info[i].len);
+	              nv_add_nvp (res, "SQL_Text", qbuf);
+
+	              free (qbuf);
+	            }
+	        }
+	    }
 	}
       nv_add_nvp (res, "close", "transaction");
     }
   nv_add_nvp (res, "close", "transactioninfo");
+
+  if (query_p)
+    {
+      free (query_p);
+    }
+
+  if (sql_info)
+    {
+      free (sql_info);
+    }
+
   fclose (infile);
   unlink (tmpfile);
   unlink (errfile);
 
   return ERR_NO_ERROR;
 }
 
+/*
+ * Find file size of output file from 'cubrid tranlist database'
+ * and number of SQL inside in it. (for calloc info)
+ */
+
+static int
+get_sql_info (char *dbmt_file, int *file_size)
+{
+  int num_queries = 0;
+  struct stat sb;
+  FILE *qry_fp;
+  char buf[4096];
+  const char *sql_id_mark = "SQL_ID:";
+  int sql_id_mark_len = strlen (sql_id_mark);
+
+  if (stat (dbmt_file, &sb) < 0)
+    {
+      return -1;
+    }
+
+  if ((qry_fp = fopen (dbmt_file, "r")) == NULL)
+    {
+      return -1;
+    }
+
+  while (fgets (buf, sizeof (buf), qry_fp))
+    {
+      if (strncmp (buf, sql_id_mark, sql_id_mark_len) == 0)
+      {
+        num_queries++;
+      }
+    }
+
+  *file_size = sb.st_size;
+  fclose (qry_fp);
+
+  return num_queries;
+}
+
+/*
+ * Mainiatin single dimension arrary to store all SQL_Text
+ * And keep offset, length info for each SQL text in TS_SQL_INFO array
+ * on query_p array.
+ */
+
+static int
+get_sql_text (char *tmpfile, char *query_p, TS_SQL_INFO *sql_info, int query_file_size)
+{
+  FILE *qry_fp;
+  int num_queries = 0;
+  int sql_index = 0;
+  int idx = 0;
+  int offset = 0;
+  int ret = -1;
+  char buf[4096];
+  char *sqlid_p;
+  char *newline_p;
+  const char *sql_id_mark = "SQL_ID:";
+  int sql_id_mark_len = strlen (sql_id_mark);
+
+
+  if ((qry_fp = fopen (tmpfile, "r")) == NULL)
+    {
+      return -1;
+    }
+
+  while (fgets (buf, sizeof (buf), qry_fp))
+    {
+      if (strncmp (buf, sql_id_mark, sql_id_mark_len) != 0)
+      {
+        continue;
+      }
+
+      sqlid_p = strchr (buf, ':');
+      if (sqlid_p == NULL || strlen (sqlid_p) < 2)
+      {
+        continue;
+      }
+
+      sqlid_p+= 2; /* example tranlit output look like: SQL_ID: 5d5807aaab63c */
+
+      newline_p = strchr (sqlid_p, '\n');
```

Comment:
newline_p is an unused variable.

Reply:
newline_p is used as a temp pointer to replace newline with '\0' (you can see a line above, #7124)

`newline_p = strchr (sqlid_p, '\n');`